File: fsharp/leap/Solution/Leap/Leap.fs
module LeapYear

let isLeapYear year = 
    let mod4 = (year % 4) = 0
    let mod100 = (year % 100)  = 0
    let mod400 = (year % 400) = 0
    match (mod4,mod100,mod400) with
    | true, true, true-> true
    | true, false, _ -> true
    | _ -> false

File: go/leap/leap.go
package leap

const testVersion = 3

// IsLeapYear test if a provided year is a leapyear
func IsLeapYear(year int) bool {
	return year%4 == 0 && (year%100 != 0 || (year%100 == 0 && year%400 == 0))
}
